joomla/joomla-cms
# [4.0] Add Copy to Clipboard button for API token

Pull Request for Issue #28995

### Summary of Changes

This PR adds a "Copy" button allowing you to copy the API token to your clipboard.

Evergreen browsers get to use the async Clipboard API, however this will only work over a secure connection (`https://`), so if you're on a insecure protocol, you will fallback to `execCommand(...)`.

### Testing Instructions

**NOTE:** The button is supposed to appear next to the input, however you'll see it appear below the input. This is due to the way Joomla is styling it's parameters. It's out of scope and I will NOT fix that in this PR, so please do not flag it as an issue.

1. Run `node build.js --compile-js` from your terminal
2. Go to `administrator/index.php?option=com_users` and click on a user
3. Navigate to the `Joomla API Token` tab
4. Click the `Copy` button

### Expected result

The token will be copied to your clipboard
